circuit-types/src/traits.rs

```diff
@@ -427,30 +248,28 @@ pub trait SecretShareType: Sized + BaseType {
 }
 
 /// Implementing types represent a secret share allocated in a constraint system
-pub trait SecretShareVarType<L: LinearCombinationLike>: Sized + CircuitVarType<L> {
+pub trait SecretShareVarType: Sized + CircuitVarType {
     /// The base type that this secret share is a representation of
-    type Base: CircuitVarType<LinearCombination>;
-    /// The type that results from blinding or unblinding the share
-    type BlindType: CircuitVarType<LinearCombination>;
```

Comment:
Why do we no longer represent the blinded/unblinded type as a separate type? Was this just not really necessary in the first place?

Reply:
They were different only for dalek-specific reasons; i.e. adding two `Variable`s created a `LinearCombination` which is a different runtime type, these separate associated types realize this at higher layers. Because `Variable` is "closed" under addition we no longer have to do this